EXILED/Exiled.API/Features/Items/Firearm.cs

```diff
@@ -105,24 +123,97 @@ public static IReadOnlyDictionary<Player, Dictionary<FirearmType, AttachmentIden
+        /// </remarks>
+        public BarrelMagazine BarrelMagazine { get; }
+
+        /// <summary>
+        /// Gets or sets the amount of ammo in the firearm magazine.
+        /// </summary>
+        public int MagazineAmmo
+        {
+            get => PrimaryMagazine.Ammo;
+            set => PrimaryMagazine.Ammo = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the amount of ammo in the firearm barrel.
         /// </summary>
-        public int Ammo
+        /// <remarks>
+        /// not working for Revolver and ParticleDisruptor.
+        /// </remarks>
+        public int BarrelAmmo
         {
-            get => (Base.Modules[Array.IndexOf(Base.Modules, typeof(MagazineModule))] as MagazineModule).AmmoStored;
-            set => (Base.Modules[Array.IndexOf(Base.Modules, typeof(MagazineModule))] as MagazineModule).AmmoStored = value;
+            get
+            {
+                return BarrelMagazine?.Ammo ?? 0;
+            }
+
+            set
+            {
+                if (BarrelMagazine != null)
+                    BarrelMagazine.Ammo = value;
+            }
         }
 
+        /// <summary>
+        /// Gets the total amount of ammo in the firearm.
+        /// </summary>
+        public int TotalAmmo =>
+            Base.GetTotalStoredAmmo();
+
         /// <summary>
         /// Gets or sets the max ammo for this firearm.
         /// </summary>
-        /// <remarks>Disruptor can't be used for MaxAmmo.</remarks>
-        public int MaxAmmo
+        public int MaxMagazineAmmo
+        {
+            get => PrimaryMagazine.MaxAmmo;
+            set => PrimaryMagazine.MaxAmmo = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the amount of max ammo in the firearm barrel.
+        /// </summary>
+        /// <remarks>
+        /// not working for Revolver and ParticleDisruptor.
+        /// </remarks>
+        public int MaxBarrelAmmo
         {
-            get => (Base.Modules[Array.IndexOf(Base.Modules, typeof(MagazineModule))] as MagazineModule).AmmoMax;
-            set => (Base.Modules[Array.IndexOf(Base.Modules, typeof(MagazineModule))] as MagazineModule)._defaultCapacity = value; // Synced?
+            get
+            {
+                return BarrelMagazine?.MaxAmmo ?? 0;
+            }
+
+            set
+            {
+                if (BarrelMagazine != null)
+                    BarrelMagazine.MaxAmmo = value;
+            }
         }
 
+        /// <summary>
+        /// Gets the total amount of ammo in the firearm.
+        /// </summary>
+        public int TotalMaxAmmo =>
+            Base.GetTotalMaxAmmo();
+
+        /// <summary>
+        /// Gets or sets a ammo drain per shoot.
+        /// </summary>
+        /// <remarks>
+        /// Always <see langword="1"/> by default.
+        /// Applied on a high layer nether basegame ammo controllers.
+        /// </remarks>
+        public int AmmoDrain { get; set; } = 2;
```

Comment:
My brain didn't get what these remarks mean and why 2 is needed

Reply:
Oh, i forgot to set it for 1 by default, i used `2` for testing😭